### PR TITLE
feat(date): #CO-1816 get statistics up to 1 year before now

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
It is now possible to check statistics at any time from 01-01-2020 to now, with a 1-year-slot maximum.

## Issue ticket number and link
[CO-1816](https://jira.support-ent.fr/browse/CO-1816)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/js/controller.js [hunk 1]
```diff
@@ -130,15 +130,8 @@ function StatisticsController($scope, template, model) {
 			return;
 		}
 
+		var minDate = moment('01-01-2020', 'DD-MM-YYYY');
```
Comment (alicedraillard) [inline src/main/resources/public/js/controller.js:133]: set minimum date to 01/01/2020

File: src/main/resources/public/js/controller.js [hunk 2]
```diff
@@ -196,6 +189,26 @@ function StatisticsController($scope, template, model) {
 		$scope.form.to = $scope.toDates[$scope.toDates.length-1].moment;
 	};
 
+	$scope.isIntervalValid = () => {
```
Comment (alicedraillard) [inline src/main/resources/public/js/controller.js:192]: check if slot is valid aka :
- start date is before end date
- slot is one year or less (12 months, so october 2022 to september 2023 for example)

File: src/main/resources/public/js/controller.js [hunk 3]
```diff
@@ -196,6 +189,26 @@ function StatisticsController($scope, template, model) {
...
+		var before = $scope.form.from.moment ? $scope.form.from.moment : $scope.form.from;
+		var after = $scope.form.to.moment ? $scope.form.to.moment : $scope.form.to;
+		var isStartBeforeEnd = before.isBefore(after);
+		var isIntervalMoreThanOneYearisIntervalMoreThanOneYear = Math.abs(before.diff(after)) < 31556926000;
+		return isStartBeforeEnd && isIntervalMoreThanOneYearisIntervalMoreThanOneYear;
+	}
+
+	$scope.initStartDate = () => {
+		// Set expectedStartDate5 to september 1st of current school year
```
Comment (alicedraillard) [inline src/main/resources/public/js/controller.js:201]: Wee keep the original view when loading the page : stats are from last september to now

File: src/main/resources/public/template/form.html [hunk 4]
```diff
@@ -67,7 +69,10 @@
 				</span>
 			</div>
 
-			<button ng-if="!form.processing" class="right-magnet" ng-click="getData()"><i18n>statistics.form.display.chart</i18n></button>
+			<div display="flex">
```
Comment (alicedraillard) [inline src/main/resources/public/template/form.html:72]: warning in case slot is not valid

File: src/main/resources/i18n/en.json [hunk 5]
```diff
@@ -20,5 +20,120 @@
     "group.school.Teacher": "Tous les enseignants de l'{0}.",
     "group.school.Guest": "Tous les invités de l'{0}.",
     "plus": "Plus",
-    "statistics.title": "Statistics"
+    "statistics.title": "Statistics",
```
Comment (alicedraillard) [inline src/main/resources/i18n/en.json:23]: match english translations to the french ones

**Review by JinDouang — APPROVED** [review]

**Pull request merged**